cosmos/lib/ante/ignore.go

```diff
@@ -18,30 +18,37 @@
 // MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
 // TITLE.
 
-package ante
+package antelib
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"pkg.berachain.dev/polaris/cosmos/x/evm/types"
 	"pkg.berachain.dev/polaris/lib/utils"
 )
 
-// EthSkipDecorator is an AnteDecorator that wraps an existing AnteDecorator. It allows
-// EthTransactions to skip said Decorator by checking the first message in the transaction
-// for an EthTransactionRequest. This is safe since EthTransactions are guaranteed to be
-// the first and only message in a transaction.
-type EthSkipDecorator[T sdk.AnteDecorator] struct {
-	decorator T
+// IgnoreDecorator is an AnteDecorator that wraps an existing AnteDecorator. It allows
+// EthTransactions to skip said Decorator by checking to see if the transaction
+// contains a message of the given type.
+type IgnoreDecorator[D sdk.AnteDecorator, M sdk.Msg] struct {
```

Comment:
Didn't know you could wrap multiple types like this. pretty sick

Reply:
yeah it's super useful, I' a premature optimization enthusiast